packages/h/hpx/xmake.lua

```diff
@@ -46,16 +60,105 @@ package("hpx")
+        end
         import("package.tools.cmake").install(package, configs)
+
+        package:add("includedirs", "include")
+        package:add("linkdirs", "lib")
+
+        if package:config("shared") then
+            -- handle shared lib
+            package:add("links", "hpx", "hpx_iostreams", "hpx_core", "hpx_init")
+        else
+            -- handle static lib
+            import("utils.binary.deplibs", {alias = "get_depend_libs"})
+            import("core.base.graph")
+
+            -- scan all libs
+            local hpx_libs_map = {}
+            local lib_pattern = package:is_plat("windows") and "*.lib" or "*.a"
+            cprint('\n${red}current lib path: %s', path.join(package:installdir(), "lib", lib_pattern))
+            for _, libpath in ipairs(os.files(path.join(package:installdir(), "lib", lib_pattern))) do
+                local basename = path.basename(libpath)
+                local linkname
+                if package:is_plat("windows") then
+                    linkname = basename
+                else
+                    linkname = basename:gsub("lib(.-)", "%1")
+                end
+                cprint('${bright} basename=%s, linkname=%s', basename, linkname)
+                hpx_libs_map[linkname] = libpath
+            end
+
+            cprint("\n*************************\n${yellow}Found HPX static libraries:")
+            for name, path in pairs(hpx_libs_map) do
+                cprint("  - ${bright green}" .. name .. "${clear} @ ${blue}" .. path)
+            end
+
+            -- create a DAG
+            local dag = graph.new()
+            local cnt = 0
+            for linkname, libpath in pairs(hpx_libs_map) do
+                -- dag:add_vertex(linkname)
+                cprint("\n\n${bright underline}deps analyzing... %s", libpath)
+                local dependencies = get_depend_libs(libpath, {plat = package:plat(), arch = package:arch()})
+                if (dependencies) then
+                    print(dependencies)
+                else
+                    cprint("${red}\n !!!status!!!: have analyzed %d lib(s)", cnt)
+                    os.raise("error!!!!!!!! found no deps of the current lib")
+                end
+                for _, dep_path in ipairs(dependencies) do
+                    local dep_basename = path.basename(dep_path)
+                    local dep_linkname
+                    if package:is_plat("windows") then
+                        dep_linkname = dep_basename:gsub("%.lib$", "")
+                    else
+                        dep_linkname = dep_basename:gsub("lib(.-)%.a", "%1")
+                    end
+                    if hpx_libs_map[dep_linkname] then
+                        cprint("  ${cyan}%s ${yellow}-> ${green}%s", linkname, dep_linkname)
+                        dag:add_edge(linkname, dep_linkname)
+                    end
+                end
+            end
+
+            local sorted_links = dag:topological_sort()
```

Comment:
if enable recursive, it will get all deps and use toposort on dev version

https://github.com/xmake-io/xmake/blob/b2628769d3206449e9301cb86bd87df9c789ccd5/xmake/modules/utils/binary/deplibs.lua#L382-L395

and topological_sort is deprecated, we should use topo_sort